lib/slist.c

```diff
@@ -65,19 +53,26 @@ struct curl_slist *Curl_slist_append_nodup(struct curl_slist *list, char *data)
 
   DEBUGASSERT(data);
 
-  new_item = malloc(sizeof(struct curl_slist));
+  if(list)
+    new_item = malloc(sizeof(struct curl_slist));
+  else
+    new_item = malloc(sizeof(struct curl_slist_head));
+
   if(!new_item)
     return NULL;
 
   new_item->next = NULL;
   new_item->data = data;
 
   /* if this is the first item, then new_item *is* the list */
-  if(!list)
+  if(!list) {
+    ((struct curl_slist_head*)new_item)->tail = new_item;
     return new_item;
+  }
 
-  last = slist_get_last(list);
+  last = ((struct curl_slist_head*)list)->tail;
```

Comment:
The `ASAN` job detects an invalid read here because the `tests/http/clients/tls-session-reuse.c` uses a stack-allocated zero-initialised `curl_slist`, and calls `curl_slist_append(&slist, string)` on it instead of just calling `curl_slist_append(NULL, string)` for some reason. :/

https://github.com/curl/curl/blob/912d80c68019d3d9a4ceb9993a596dff8009f4d0/tests/http/clients/tls-session-reuse.c#L188

This patch may be backward incompatible, though that test file is fairly recent (5 months old), and it is the only instance of that pattern in the `curl` repo:
```console
$ git grep 'struct[^*]*curl_slist[^*]*;'
lib/slist.c:68:  new_item = malloc(sizeof(struct curl_slist));
tests/http/clients/tls-session-reuse.c:188:  struct curl_slist resolve;
```